server/lib/pbench/process_tb.py

```diff
@@ -0,0 +1,113 @@
+import os
+import subprocess
+
+from collections import namedtuple
+from logging import Logger
+from pathlib import Path
+
+from pbench import PbenchConfig
+
+Results = namedtuple("Results", ["nstatus", "ntotal", "ntbs"])
+
+
+class ProcessTb:
+    def __init__(self, config: PbenchConfig, logger: Logger):
+        """ Processes Tar ball, gets the tar ball from receiving directory
+        and copies it to the remote server by creating a subprocess which
+        call Agent's 'pbench-results-push' command.
+
+            Args -
+                config -- PbenchServer config object
+                logger -- logger object to use when emitting log entries during
+                            operation
+
+        """
+        self.config = config
+        self.logger = logger
+        # receive_dir -- directory where tarballs are received from agent
+        self.receive_dir = self.get_receive_dir()
+
+    def get_receive_dir(self) -> Path:
```

Comment:
Consider changing this name to be preceded by an underscore (`_`) to indicate that the method is not public, just a helper function for the class itself.